handler/search_test.go

```diff
@@ -141,4 +146,108 @@ func Test(t *testing.T) {
 			})
 		})
 	})
+
+
+	Convey("Given a search request", t, func() {
+
+		recorder := httptest.NewRecorder()
+		requestBodyReader := bytes.NewReader([]byte("{not a valid document}"))
+		request, _ := http.NewRequest("GET", "/search?q=armed", requestBodyReader)
+
+		Convey("When the query handler is called", func() {
+
+			mockSearchClient := searchtest.NewMockSearchClient()
+			handler.SearchClient = mockSearchClient
+			handler.Search(recorder, request)
+
+			Convey("Then the search client is called with the expected parameters", func() {
+				So(recorder.Code, ShouldEqual, http.StatusOK)
+				So(mockSearchClient.QueryRequests[0], ShouldEqual, "armed")
+			})
+		})
+	})
+
+	Convey("Given a suggest request", t, func() {
+
+		recorder := httptest.NewRecorder()
+		requestBodyReader := bytes.NewReader([]byte("{not a valid document}"))
+		request, _ := http.NewRequest("GET", "/suggest?q=armed", requestBodyReader)
+
+		Convey("When the query handler is called", func() {
+
+			mockSearchClient := searchtest.NewMockSearchClient()
+			mockSearchClient.CustomQueryFunc = func(term string, index string) (*model.SearchResponse, error) {
+				return mockSuggestResponse, nil
+			}
+			handler.SearchClient = mockSearchClient
+			handler.Suggest(recorder, request)
+
+			Convey("Then the response contains the content returned from the search client.", func() {
+				So(recorder.Code, ShouldEqual, http.StatusOK)
+
+				actualResponse := &model.SearchResponse{}
+				_ = json.Unmarshal(recorder.Body.Bytes(), actualResponse)
+
+				So(actualResponse.TotalResults, ShouldEqual, mockSearchResponse.TotalResults)
+				So(actualResponse.Results[0].ID, ShouldEqual, mockSearchResponse.Results[0].ID)
+				So(actualResponse.Results[1].ID, ShouldEqual, mockSearchResponse.Results[1].ID)
```

Comment:
This should be comparing against the mockSuggestResponse - although they are currently the same 😄